Fixed compiler warning.

GeminiServer opened the pull request:
'void operator delete(void*, std::size_t)' called on unallocated object 'result' 
.clang-format adapted to work with macos

GeminiServer committed:
Fixed compiler warning.

'void operator delete(void*, std::size_t)' called on unallocated object 'result'
.clang-format adapted to work with macos

GeminiServer commented:
Can someone please check this and merge. Thanks!

thewhobox commented on src/OpenKNX/Time/TimeManager.cpp:
```diff
@@ -508,7 +508,7 @@ namespace OpenKNX
             else
                 result += "366,367"; // Always standard time
-            return result;
+            return result.c_str();
```
Die Funktion hat doch als Rückgabewert std::string und response ist ebenfalls einer.
Wieso wird hier in ein char Array konvertiert?

thewhobox replied:
```C++
std::string result = "";
```
Wäre das auch eine Lösung, damit result allociert wird?

GeminiServer replied:
Eher nicht. buildTimezoneString hat als Rückgabewert std::string und response ist ebenfalls ein std::string. Der Grund, warum hier char-Array zurückgegeben wird, liegt daran, dass std::string in dieser Funktion unnötig ist. Ein const char* hätte hier völlig ausgereicht, wenn man sich die Verwendung in common anschaut. Das wäre die sauberste Lösung.

std::string result = ""; wäre keine Lösung. Wir können nicht vermeiden, dass die externe Bibliothek versucht, ein dealloc auf das zurückgegebene std::string-Objekt durchzuführen. Daher ist es besser, const char* zu verwenden. Deswegen kommt auch die Meldung. 

@mgeramb, das ist doch kein Code, oder? Die Warnung tritt nur in Verbindung mit dem Target Pi PICO auf. Beim ESP Target gibt es hiermit kein Problem. Was hältst du davon, wenn wir das hier auf const char* TimeManager::buildTimezoneString umstellen? Dann sollte es sauber gefixt sein.

GeminiServer replied:
Siehe mein letzen commit! So sollte es besser sein! Ist aber dann ein Refactoring. Sowas mache ich ungern... schaut bitte drüber! Danke.

traxanos replied:
ein const char* kann man hier nicht machen, da der rückgabewert in der function erst allociert wird. sprich bei rückgabe ist dieser schon wieder freigegeben. der zugriff ist also glpcksssache. daher nutzt man eben std::string um genau sowas zu vermeiden.

GeminiServer committed:
Refactor buildTimezoneString to return const char*

- Changed the return type of `buildTimezoneString` from `std::string` to `const char*`.
- Removed unnecessary conversion to `std::string` and used `const char*` instead.
- Ensured proper handling of the timezone string and daylight saving mode.
- This change resolves the issue with the external library attempting to deallocate the returned `std::string` object. (Target Pi PICO builds!)
- The warning related to deallocation on unallocated object is now resolved.

traxanos commented on src/OpenKNX/Time/TimeManager.cpp:
```diff
@@ -416,7 +416,9 @@ namespace OpenKNX
             // <Enumeration Text="Nouméa (+12 Stunden)" Value="11" Id="%ENID%" />
             // <Enumeration Text="Wellington (+12 Stunden)" Value="12" Id="%ENID%" />
 
+            static char result[100];                                                 // Puffer für das Ergebnis
```
ein static innerhalb einer cpp class ist auch eher untypisch. soll es  global reserviert werden, dann gehört das in die header definition rein. aber wie schon im mr geschrieben ist der weg hier sowieso falsch.

traxanos replied:
jetzt wo ich nochmal drüber nachdenke, das static verhindert natürlich das der wert bei rückgabe nicht freigegeben wird. müsste also tatsächlich klappen. aber warum ist man gegen std::string. wir bauen hier doch hier cpp. da muss man sich doch nicht rumärgern wie in c. schon wieder pauschal 100 bytes dauerhaft im ram ist auch sch...

mgeramb commented:
@GeminiServer ich bin über die static Implementierung nicht glücklich. Könntest du bitte testen ob die Version bei dir geht: https://github.com/OpenKNX/OGM-Common/pull/83
@traxanos

mgeramb commented:
Fixed with https://github.com/OpenKNX/OGM-Common/pull/83
